WordPress/Classes/Utility/Analytics/AnalyticsTrackerAutomatticTracks.swift

```diff
@@ -0,0 +1,241 @@
+import Foundation
+import AutomatticTracks
+import WordPressShared
+import BuildSettingsKit
+
+@objc public final class AnalyticsTrackerAutomatticTracks: NSObject, WPAnalyticsTracker {
+    private let contextManager: TracksContextManager
+    private let tracksService: TracksService
+    private var userProperties: [String: Any] = [:]
+    private var appURLScheme: String
+    private var cachedAnonymousUserID: String?
+    private var cachedCurrentUserID: String?
+
+    @objc convenience public override init() {
+        let settings = BuildSettings.current
+        self.init(
+            eventNamePrefix: WPAnalyticsTesting.eventNamePrefix ?? settings.eventNamePrefix,
+            platform: WPAnalyticsTesting.explatPlatform ?? settings.explatPlatform,
+            appURLScheme: settings.appURLScheme
+        )
+    }
+
+    init(
+        eventNamePrefix: String,
+        platform: String,
+        appURLScheme: String
+    ) {
+        contextManager = TracksContextManager()
+        tracksService = TracksService(contextManager: contextManager)
+        tracksService.eventNamePrefix = eventNamePrefix
+        tracksService.platform = platform
+        self.appURLScheme = appURLScheme
+    }
+
+    // MARK: - WPAnalyticsTracker
+
+    public func track(_ stat: WPAnalyticsStat) {
+        track(stat, withProperties: nil)
+    }
+
+    public func track(_ stat: WPAnalyticsStat, withProperties properties: [AnyHashable: Any]?) {
+        guard let event = TracksMappedEvent.make(for: stat) else {
+            DDLogInfo("WPAnalyticsStat not supported by AnalyticsTrackerAutomatticTracks: \(stat)")
+            return
+        }
+
+        var mergedProperties: [AnyHashable: Any] = event.properties ?? [:]
+        for (key, value) in properties ?? [:] {
+            mergedProperties[key] = value
+        }
```

Comment:
Nitpick: The above should be able to be implemented in one line.
```
let mergedProperties = (event.properties ?? [:]).merging(properties ?? [:]) { (_, new) in new }
```

Reply:
Updated. I'm not sure why I keep avoiding `merging`. It is more clear and less code this way.